remoting/subsystem/src/main/java/org/jboss/as/remoting/ConnectorUtils.java

```diff
@@ -156,16 +156,20 @@ private static void addSasl(OperationContext context, ModelNode sasl, OptionMap.
     private static void addOptions(OperationContext context, ModelNode properties, OptionMap.Builder builder) throws OperationFailedException {
         final ClassLoader loader = WildFlySecurityManager.getClassLoaderPrivileged(ConnectorUtils.class);
         for (Property property : properties.asPropertyList()) {
-            String name = property.getName();
-            if (!name.contains(".")) {
-                name = "org.xnio.Options." + name;
-            }
-            final Option option = Option.fromString(name, loader);
+            final Option option = getAndValidateOption(property.getName());
             String value = PropertyResource.VALUE.resolveModelAttribute(context, property.getValue()).asString();
             builder.set(option, option.parseValue(value, loader));
         }
     }
 
+    static Option<?> getAndValidateOption(String name) {
```

Comment:
method should be private and should pass the `ClassLoader` in parameter instead of calling `WildFlySecurityManager.getClassLoaderPrivileged`every time.

Reply:
@jmesnil I have added passing ClassLoader as a parameter but it meant to add it also in ConnectorChildResource. And that's the other thing. Because of ConnectorChildResource it cannot be private